File: BE/Src/MIT.Fwk.Infrastructure/Entities/GeoSecondDivision.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using MIT.Fwk.Core.Attributes;
using System.Collections.Generic;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [SkipJwtAuthentication]
    [SkipClaimsValidation]
    [SkipRequestLogging]
    public class GeoSecondDivision : Identifiable<int>
    {
        [Attr]
        public string Name { get; set; }
        [Attr]
        public int GeoFirstDivisionId { get; set; }
        [Attr]
        public int GeoCountryId { get; set; }
        [HasOne]
        public virtual GeoFirstDivision GeoFirstDivision { get; set; }
        [HasOne]
        public virtual GeoCountry GeoCountry { get; set; }
        [HasMany]
        public virtual ICollection<GeoThirdDivision> GeoThirdDivisions { get; set; }
        [HasMany]
        public virtual ICollection<GeoMapping> GeoMappings { get; set; }
    }
}

File: FE/WEB/app/models/geo-second-division.js
import Model, { attr, belongsTo, hasMany } from '@ember-data/model';

export default class GeoSecondDivisionModel extends Model {
  @belongsTo('geo-country', { async: true, inverse: 'geoSecondDivisions' })
  geoCountry;
  @hasMany('geo-first-division', { async: true, inverse: 'GeoSecondDivisions' })
  GeoFirstDivision;
  @hasMany('geo-third-division', { async: true, inverse: 'GeoSecondDivisions' })
  GeoThirdDivisions;
  @hasMany('geo-mapping', { async: true, inverse: 'GeoSecondDivision' })
  geoMappings;

  @attr('string', { defaultValue: '' }) name;
  @attr('number', { defaultValue: 0 }) geoCountryId;
  @attr('number', { defaultValue: 0 }) geoFirstDivisionId;
}
